runtime/indent/sh.vim

```diff
@@ -203,7 +204,7 @@ function! s:is_function_definition(line)
 endfunction
 
 function! s:is_array(line)
-  return a:line =~ '^\s*\<\k\+\>=('
+  return a:line =~ '^\s*\(local\s\+\)\?\<\k\+\>=('
```

Comment:
```suggestion
  return a:line =~ '^\s*\(\(declare\|typeset\|local\)\s\+\(-[Aalrtu]\+\s\+\)\?\)\?\<\k\+\>=('
  " TODO: A and a are mutually exclusive
```

`-a`/`-A` of `declare`/`typeset`/`local` respectively specifies that the variables are indexed/associative arrays, but is not mandatory; plus, further options like `-r` (readonly) can be added:

```bash
declare -au x=(foo bar) y=(baz qux)  # `-u` for uppercase
typeset -r z=(foobar)                # `-a` is omitted
```

Reply:
Hi,

LGTM

i was able to reproduce the issues with declare and typeset that you mentioned, so i will add tests for these and probably commit your suggestion soon =)

Thanks =)